{"pr_number": 1365, "pr_title": "Return default token endpoint for issuer ID in the tenant-qualified URLs mode", "pr_author": "chamathns", "pr_createdAt": "2020-04-24T10:29:51Z", "pr_url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365", "timeline": [{"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ3NTk4Mg==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414475982", "body": "This shouldn't be a runtime exception right?\r\nShall we throw a IdentityOAuth2Exception with a proper error message and including the original exception", "bodyText": "This shouldn't be a runtime exception right?\nShall we throw a IdentityOAuth2Exception with a proper error message and including the original exception", "bodyHTML": "<p dir=\"auto\">This shouldn't be a runtime exception right?<br>\nShall we throw a IdentityOAuth2Exception with a proper error message and including the original exception</p>", "author": "mefarazath", "createdAt": "2020-04-24T10:39:42Z", "path": "components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java", "diffHunk": "@@ -3127,14 +3127,22 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String\n \n     public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2Exception {\n \n-        IdentityProvider identityProvider = getResidentIdp(tenantDomain);\n-        FederatedAuthenticatorConfig[] fedAuthnConfigs = identityProvider.getFederatedAuthenticatorConfigs();\n-        // Get OIDC authenticator\n-        FederatedAuthenticatorConfig oidcAuthenticatorConfig =\n-                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,\n-                        IdentityApplicationConstants.Authenticator.OIDC.NAME);\n-        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),\n-                IDP_ENTITY_ID).getValue();\n+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {\n+            try {\n+                return ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL).build().getAbsolutePublicURL();\n+            } catch (URLBuilderException e) {\n+                throw new OAuthRuntimeException(\"Error while building url for context: \" + OAUTH2_TOKEN_EP_URL);", "originalCommit": "5bfee42bc134416e7d6e1fa7c22bb9b45b138692", "replyToReviewId": null, "replies": [{"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ4MzgyOQ==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414483829", "bodyText": "Yes. The PR is now updated.", "author": "chamathns", "createdAt": "2020-04-24T10:54:01Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ3NTk4Mg=="}], "type": "inlineReview"}, {"oid": "11277d223da66490be57b0a7d18a28bb209702e4", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/11277d223da66490be57b0a7d18a28bb209702e4", "message": "Return default token endpoint for issuer ID in the tenant-qualified URLs mode", "committedDate": "2020-04-24T10:53:00Z", "type": "commit"}, {"oid": "11277d223da66490be57b0a7d18a28bb209702e4", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/11277d223da66490be57b0a7d18a28bb209702e4", "message": "Return default token endpoint for issuer ID in the tenant-qualified URLs mode", "committedDate": "2020-04-24T10:53:00Z", "type": "forcePushed"}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ5NDI5NA==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414494294", "body": "I think it would be better say \"Error while retrieving ID Token issuer value for tenantDomain: \" + tenantDomain\r\nhere. \r\n\r\nCan you check how the exception thrown is handled? If we are handling and logging a similar message as I suggested above we do not need to change.", "bodyText": "I think it would be better say \"Error while retrieving ID Token issuer value for tenantDomain: \" + tenantDomain\nhere.\nCan you check how the exception thrown is handled? If we are handling and logging a similar message as I suggested above we do not need to change.", "bodyHTML": "<p dir=\"auto\">I think it would be better say \"Error while retrieving ID Token issuer value for tenantDomain: \" + tenantDomain<br>\nhere.</p>\n<p dir=\"auto\">Can you check how the exception thrown is handled? If we are handling and logging a similar message as I suggested above we do not need to change.</p>", "author": "mefarazath", "createdAt": "2020-04-24T11:13:07Z", "path": "components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java", "diffHunk": "@@ -3127,14 +3127,23 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String\n \n     public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2Exception {\n \n-        IdentityProvider identityProvider = getResidentIdp(tenantDomain);\n-        FederatedAuthenticatorConfig[] fedAuthnConfigs = identityProvider.getFederatedAuthenticatorConfigs();\n-        // Get OIDC authenticator\n-        FederatedAuthenticatorConfig oidcAuthenticatorConfig =\n-                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,\n-                        IdentityApplicationConstants.Authenticator.OIDC.NAME);\n-        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),\n-                IDP_ENTITY_ID).getValue();\n+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {\n+            try {\n+                return ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL).build().getAbsolutePublicURL();\n+            } catch (URLBuilderException e) {\n+                String errorMsg = String.format(\"Error while building url for '%s' context\", OAUTH2_TOKEN_EP_URL);", "originalCommit": "11277d223da66490be57b0a7d18a28bb209702e4", "replyToReviewId": null, "replies": [{"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ5OTY2MQ==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414499661", "bodyText": "It only has \"Error while retrieving OIDC Id token issuer\" + e. No mention of a tenant domain. Should we add the tenant domain there?", "author": "chamathns", "createdAt": "2020-04-24T11:22:50Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ5NDI5NA=="}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDUwNDQxMQ==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414504411", "bodyText": "Okay let add it there", "author": "mefarazath", "createdAt": "2020-04-24T11:31:26Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ5NDI5NA=="}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDUwNDgzNw==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414504837", "bodyText": "Also lets try to include the tenantDomain here too", "author": "mefarazath", "createdAt": "2020-04-24T11:32:13Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ5NDI5NA=="}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDUxMTE1Nw==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414511157", "bodyText": "Updated the PR with the change", "author": "chamathns", "createdAt": "2020-04-24T11:43:28Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDQ5NDI5NA=="}], "type": "inlineReview"}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDUzMjc0NQ==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414532745", "body": "Can we present this message more clearly?\r\n\r\n```suggestion\r\n                String errorMsg = String.format(\"Error while building url  '%s' for tenant domain: %s\",\r\n```", "bodyText": "Can we present this message more clearly?\n  \n    \n      \n        Suggested change\n        \n          \n    \n\n        \n      \n    \n    \n      \n          \n            \n                            String errorMsg = String.format(\"Error while building url for '%s' context for tenant domain: %s\",\n          \n          \n            \n                            String errorMsg = String.format(\"Error while building url  '%s' for tenant domain: %s\",", "bodyHTML": "<p dir=\"auto\">Can we present this message more clearly?</p>\n  <div class=\"my-2 border rounded-1 js-suggested-changes-blob diff-view js-check-bidi\" id=\"\">\n    <div class=\"f6 p-2 lh-condensed border-bottom d-flex\">\n      <div class=\"flex-auto flex-items-center color-fg-muted\">\n        Suggested change\n        <span class=\"tooltipped tooltipped-multiline tooltipped-s\" aria-label=\"This code change can be committed by users with write permissions.\">\n          <svg aria-hidden=\"true\" height=\"16\" viewBox=\"0 0 16 16\" version=\"1.1\" width=\"16\" data-view-component=\"true\" class=\"octicon octicon-info hide-sm\">\n    <path fill-rule=\"evenodd\" d=\"M8 1.5a6.5 6.5 0 100 13 6.5 6.5 0 000-13zM0 8a8 8 0 1116 0A8 8 0 010 8zm6.5-.25A.75.75 0 017.25 7h1a.75.75 0 01.75.75v2.75h.25a.75.75 0 010 1.5h-2a.75.75 0 010-1.5h.25v-2h-.25a.75.75 0 01-.75-.75zM8 6a1 1 0 100-2 1 1 0 000 2z\"></path>\n</svg>\n        </span>\n      </div>\n    </div>\n    <div itemprop=\"text\" class=\"blob-wrapper data file\" style=\"margin: 0; border: none; overflow-y: visible; overflow-x: auto;\">\n      <table class=\"d-table tab-size mb-0 width-full\" data-paste-markdown-skip=\"\">\n          <tbody><tr class=\"border-0\">\n            <td class=\"blob-num blob-num-deletion text-right border-0 px-2 py-1 lh-default\" data-line-number=\"\"></td>\n            <td class=\"border-0 px-2 py-1 blob-code-inner blob-code-deletion js-blob-code-deletion blob-code-marker-deletion\">                <span class=\"pl-smi\">String</span> errorMsg <span class=\"pl-k\">=</span> <span class=\"pl-smi\">String</span><span class=\"pl-k\">.</span>format(<span class=\"pl-s\"><span class=\"pl-pds\">\"</span>Error while building url <span class=\"x x-first x-last\">for</span> '%s'<span class=\"x x-first x-last\"> context</span> for tenant domain: %s<span class=\"pl-pds\">\"</span></span>,</td>\n          </tr>\n          <tr class=\"border-0\">\n            <td class=\"blob-num blob-num-addition text-right border-0 px-2 py-1 lh-default\" data-line-number=\"\"></td>\n            <td class=\"border-0 px-2 py-1 blob-code-inner blob-code-addition js-blob-code-addition blob-code-marker-addition\">                <span class=\"pl-smi\">String</span> errorMsg <span class=\"pl-k\">=</span> <span class=\"pl-smi\">String</span><span class=\"pl-k\">.</span>format(<span class=\"pl-s\"><span class=\"pl-pds\">\"</span>Error while building url  '%s' for tenant domain: %s<span class=\"pl-pds\">\"</span></span>,</td>\n          </tr>\n      </tbody></table>\n    </div>\n    <div class=\"js-apply-changes\"></div>\n  </div>\n", "author": "denuwanthi", "createdAt": "2020-04-24T12:21:05Z", "path": "components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java", "diffHunk": "@@ -3127,14 +3127,24 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String\n \n     public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2Exception {\n \n-        IdentityProvider identityProvider = getResidentIdp(tenantDomain);\n-        FederatedAuthenticatorConfig[] fedAuthnConfigs = identityProvider.getFederatedAuthenticatorConfigs();\n-        // Get OIDC authenticator\n-        FederatedAuthenticatorConfig oidcAuthenticatorConfig =\n-                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,\n-                        IdentityApplicationConstants.Authenticator.OIDC.NAME);\n-        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),\n-                IDP_ENTITY_ID).getValue();\n+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {\n+            try {\n+                return ServiceURLBuilder.create().addPath(OAUTH2_TOKEN_EP_URL).build().getAbsolutePublicURL();\n+            } catch (URLBuilderException e) {\n+                String errorMsg = String.format(\"Error while building url for '%s' context for tenant domain: %s\",", "originalCommit": "1e5a0c8916166a392cb62521d7bcddb459bc2c40", "replyToReviewId": null, "replies": [{"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDUzNzUxMg==", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1365#discussion_r414537512", "bodyText": "Updated with the changes.", "author": "chamathns", "createdAt": "2020-04-24T12:28:43Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDQxNDUzMjc0NQ=="}], "type": "inlineReview"}, {"oid": "5654c3971e0ccb838191e1bf6df56a404df617d3", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/5654c3971e0ccb838191e1bf6df56a404df617d3", "message": "Update error messages", "committedDate": "2020-04-24T12:25:53Z", "type": "commit"}, {"oid": "5654c3971e0ccb838191e1bf6df56a404df617d3", "url": "https://github.com/wso2-extensions/identity-inbound-auth-oauth/commit/5654c3971e0ccb838191e1bf6df56a404df617d3", "message": "Update error messages", "committedDate": "2020-04-24T12:25:53Z", "type": "forcePushed"}]}